app/models/order_item.rb

```diff
@@ -1,16 +1,31 @@
 class OrderItem < ActiveRecord::Base
   belongs_to :book
   belongs_to :order
+  belongs_to :cart
 
   before_save :set_unit_price_and_total
 
   validates :quantity, presence: true, numericality: { greater_than: 0 }
-  validates :book, :order, presence: true
+  validates :book, presence: true
+
+  delegate :title, to: :book
+
+  def unit_price
```

Comment:
Don't worry about this for now since we just talked about it and a fix will be coming in a later pr.